File: java/CountAll.java
import java.util.Scanner;

public class CountAll {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        System.out.print("Enter text: ");
        String text = scanner.nextLine();
        scanner.close();

        int space=0,digit=0,upperCase=0,lowerCase=0,symbol=0;
        for(int i = 0; i < text.length(); i++){
            final char ch = text.charAt(i);
            if(Character.isWhitespace(ch)){
                space++;
            }
            else if(Character.isDigit(ch)){
                digit++;
            }else if(Character.isUpperCase(ch)){
                upperCase++;
            }else if(Character.isLowerCase(ch)){
                lowerCase++;
            }else{
                symbol++;
            }
        }
        System.out.println("Number of space: " + space);
        System.out.println("Number of digit: " + digit);
        System.out.println("Number of upper case: " + upperCase);
        System.out.println("Number of lower case: " + lowerCase);
        System.out.println("Number of Special Character: " + symbol);
    }
}

File: kt/CountAll.kt
import java.util.Scanner

fun main() {
    
    var scanner = Scanner(System.`in`)
    print("Enter text: ")
    var text: String = scanner.nextLine()
    

    var space = 0
    var digit = 0
    var upperCase = 0
    var lowerCase = 0
    var symbol = 0
    for(i in 0 until text.length){
        val ch = text.toCharArray()[i]
        if(Character.isWhitespace(ch)){
            space++
        }
        else if(Character.isDigit(ch)){
            digit++
        }else if(Character.isUpperCase(ch)){
            upperCase++
        }else if(Character.isLowerCase(ch)){
            lowerCase++
        }else{
            symbol++
        }
    }
    println("Number of space: " + space)
    println("Number of digit: " + digit)
    println("Number of upper case: " + upperCase)
    println("Number of lower case: " + lowerCase)
    println("Number of Special Character: " + symbol)
}